rust/relay/src/server.rs

```diff
@@ -933,35 +965,16 @@ where
         tracing::info!(target: "relay", channel = %chan.value(), %client, %peer, %allocation, "Channel binding is now deleted (and can be rebound)");
     }
 
-    fn make_debug_error_response(
-        &mut self,
-        error_code: impl Into<ErrorCode>,
-        request: &impl StunRequest,
-    ) -> Message<Attribute> {
-        let error_code = error_code.into();
-
-        tracing::debug!(target: "relay", "{} failed: {}", request.method(), error_code.reason_phrase());
-
-        self.make_error_response(error_code, request)
-    }
-
-    fn make_warn_error_response(
+    fn make_error_response(
         &mut self,
         error_code: impl Into<ErrorCode>,
         request: &impl StunRequest,
+        error_level: ResponseErrorLevel,
     ) -> Message<Attribute> {
         let error_code = error_code.into();
 
-        tracing::warn!(target: "relay", "{} failed: {}", request.method(), error_code.reason_phrase());
-
-        self.make_error_response(error_code, request)
-    }
+        error_level.log(request.method(), error_code.reason_phrase());
```

Comment:
I find dispatching to this a bit odd, why not inline this function?

Reply:
no strong reason really, just thought it would be less intrusive in the relevant function like this. I can change it if we keep it.